test/show_methods.jl

```diff
@@ -0,0 +1,61 @@
+# Define mock structs at top level
+struct MockIndexer
```

Comment:
I agree; mocking the indexer and searcher won't be a good idea with the current design, and changing the design is not a good enough motivation to test show methods.